app/src/main/java/org/mozilla/fenix/settings/address/AddressManagementFragment.kt

```diff
@@ -0,0 +1,164 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.concept.storage.Address
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.StoreProvider
+import org.mozilla.fenix.compose.list.IconListItem
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.address.controller.DefaultAddressManagementController
+import org.mozilla.fenix.settings.address.interactor.AddressManagementInteractor
+import org.mozilla.fenix.settings.address.interactor.DefaultAddressManagementInteractor
+import org.mozilla.fenix.settings.autofill.AutofillAction
+import org.mozilla.fenix.settings.autofill.AutofillFragmentState
+import org.mozilla.fenix.settings.autofill.AutofillFragmentStore
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Displays a list of saved addresses.
+ */
+class AddressManagementFragment : Fragment() {
+
+    private lateinit var store: AutofillFragmentStore
+    private lateinit var interactor: AddressManagementInteractor
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        store = StoreProvider.get(this) {
+            AutofillFragmentStore(AutofillFragmentState())
+        }
+
+        interactor = DefaultAddressManagementInteractor(
+            controller = DefaultAddressManagementController(
+                navController = findNavController()
+            )
+        )
+
+        loadAddresses()
+
+        return ComposeView(requireContext()).apply {
+            setContent {
+                FirefoxTheme {
+                    val addresses = store.observeAsComposableState { state -> state.addresses }
+
+                    AddressList(
+                        addresses = addresses.value ?: emptyList(),
+                        onAddressClick = interactor::onSelectAddress,
+                        onAddAddressButtonClick = interactor::onAddAddressButtonClick
+                    )
+                }
+            }
+        }
+    }
+
+    /**
+     * Fetches all the addresses from the autofill storage and updates the
+     * [AutofillFragmentStore] with the list of addresses.
+     */
+    private fun loadAddresses() {
+        lifecycleScope.launch {
+            val addresses = requireContext().components.core.autofillStorage.getAllAddresses()
+
+            lifecycleScope.launch(Dispatchers.Main) {
+                store.dispatch(AutofillAction.UpdateAddresses(addresses))
+            }
+        }
+    }
+}
+
+/**
+ * A list of addresses.
+ *
+ * @param addresses A list of [Address] to display.
+ * @param onAddressClick Invoked when the user clicks on an address.
+ * @param onAddAddressButtonClick Invoked when the user clicks on the "Add address" button.
+ */
+@Composable
+fun AddressList(
```

Comment:
if we'll never use this outside of the fragment, it could optionally be made `private`

Reply:
Let me refactor this.